internal/flags.go

```diff
@@ -75,6 +75,23 @@ func objectKindSupportsLabelsFlag(kind manifest.Kind) bool {
 	return ok
 }
 
+var sloSupportingKinds = map[manifest.Kind]struct{}{
+	manifest.KindBudgetAdjustment: {},
+}
+
+func objectKindSupportsProjectSloFlag(kind manifest.Kind) bool {
+	_, ok := sloSupportingKinds[kind]
+	return ok
+}
+
+func registerProjectSloFlag(cmd *cobra.Command, storeSloIn, storeProjectIn *string) {
```

Comment:
see my comment on extender, this can be all safely moved out from here to `internal/get.go` and packed into dedicated command extender